catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -0,0 +1,183 @@
+import ddf.catalog.source.UnsupportedQueryException;
+
+/**
+ * Implementation of the {@link org.codice.ddf.migration.Migratable} interface used to migrate
+ * the current catalog provider's {@link Metacard}s.
+ */
+public class CatalogMigratableImpl extends AbstractMigratable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CatalogMigratableImpl.class);
+
+    private final CatalogProvider provider;
+
+    private final FilterBuilder filterBuilder;
+
+    private final MigrationFileWriter fileWriter;
+
+    private final CatalogMigratableConfig config;
+
+    /**
+     * Constructor.
+     *
+     * @param description   description of this migratable
+     * @param provider      provider used to retrieve the metacards to export
+     * @param filterBuilder builder used to create query filters
+     * @param fileWriter    object used to write metacards to disk
+     * @param config        export configuration information
+     */
+    public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogProvider provider,
+            @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
+            @NotNull CatalogMigratableConfig config) {
+        super(description, true);
+
+        notNull(provider, "CatalogProvider cannot be null");
+        notNull(filterBuilder, "FilterBuilder cannot be null");
+        notNull(fileWriter, "FileWriter cannot be null");
+        notNull(config, "Configuration object cannot be null");
+
+        this.provider = provider;
+        this.filterBuilder = filterBuilder;
+        this.fileWriter = fileWriter;
+        this.config = config;
+    }
+
+    /**
+     * Exports all the metacards currently stored in the catalog provider.
+     * <p/>
+     * {@inheritDoc}
+     */
+    public MigrationMetadata export(Path exportPath) throws MigrationException {
+        config.setExportPath(exportPath.resolve("catalog"));
+        MigrationTaskManager taskManager = createTaskManager(config, createExecutorService(config));
+
+        Collection<MigrationWarning> warnings = new ArrayList<>();
+        Map<String, Serializable> props = new HashMap<>();
+        Filter dumpFilter = filterBuilder.attribute(Metacard.ANY_TEXT)
+                .is()
+                .like()
+                .text("*");
+
+        useNativeQueryMode(props);
+
+        QueryImpl exportQuery = createQuery(dumpFilter);
+        QueryRequest exportQueryRequest = new QueryRequestImpl(exportQuery, props);
+
+        exportQuery.setPageSize(config.getExportPageSize());
+        exportQuery.setRequestsTotalResultsCount(false);
+
+        long exportGroupCount = 1;
+        SourceResponse response;
+
+        try {
+            List<Result> results;
+            taskManager.exportSetup();
+
+            do {
+                response = provider.query(exportQueryRequest);
+                results = response.getResults();
+
+                if (results.size() > 0) {
```

Comment:
Though, we shouldn't be using exceptions for flow control. Not when they are avoidable. Even though we're just rethrowing the exception (so i'm not sure if the performance impact is negligble) it still feels wrong